### PR TITLE
Add BloomFilter to levelDB

### PR DESCRIPTION
**Pull request by Qiao-Jin**
This PR adds BloomFIlter option and corresponding configuration into module levelDB.

### CODE REVIEW

**Comment by shargon** [comment]
Looks good, did you test the speed benefits?

**Comment by Qiao-Jin** [comment]
> Looks good, did you test the speed benefits?

Have tested in [#1599](https://github.com/neo-project/neo/pull/1599) as an alternative, I will provide detailed result later

**Comment by erikzhang** [comment]
Leveldb will create bloom filter by default.

**Comment by Tommo-L** [comment]
> Leveldb will create bloom filter by default.



We need to config `BloomFilterBitsPerKey` parameter.

File: src/LevelDBStore/LevelDBStore/config.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
-﻿{
+{
   "PluginConfiguration": {
-    "Path": "Data_LevelDB_{0}"
+    "Path": "Data_LevelDB_{0}",
+    "BloomFilterBitsPerKey": 20
```
Comment (erikzhang) [inline src/LevelDBStore/LevelDBStore/config.json:4]: I think this can be hardcoded.

Comment (Qiao-Jin) [reply src/LevelDBStore/LevelDBStore/config.json:4]: Maybe we should keep this modifible by user due to different hardware / node usage?

Comment (erikzhang) [reply src/LevelDBStore/LevelDBStore/config.json:4]: Hardware?

Comment (Qiao-Jin) [reply src/LevelDBStore/LevelDBStore/config.json:4]: No, justing depends on data. Maybe some TPS tests are needed to make sure of this figure if it should be hard-coded instead of tunable?

**Comment by Tommo-L** [comment]
Merge?

File: src/LevelDBStore/LevelDBStore/config.json (hunk 1 above)
Comment (erikzhang) [reply src/LevelDBStore/LevelDBStore/config.json:4]: Find a best value and fix it.

Comment (vncoelho) [reply src/LevelDBStore/LevelDBStore/config.json:4]: I agree with @erikzhang.
@Qiao-Jin, I believe it may be possible to even adjust it during running time in order to help the search for the best. You can slightly keep changing and tracking it during execution. For example:

* Increase in 1 unit
* Decrease in 1 unit
* Increase in 5 units
* Decrease in 5 units
* Increase in 10 units
* Decrease in 10 units

This is a like a Variable Neighborhood Search.
You slightly modify the bloom filter and observe the results, if not good you revert.
Basically you need an average value and some observation after change.

On the real code I believe that it is not possible because the microbenchmark is costly.

Comment (shargon) [reply src/LevelDBStore/LevelDBStore/config.json:4]: @Tommo-L If we resolve this conversation we can merge, the code looks good to me

Comment (Tommo-L) [reply src/LevelDBStore/LevelDBStore/config.json:4]: Okay, we'll add the benchmark with it.

Comment (Qiao-Jin) [reply src/LevelDBStore/LevelDBStore/config.json:4]: After testing it seems that bitsPeyKey = 15 is a good value to choose, with highest TPS.

**Comment by Qiao-Jin** [comment]
> Hard-code, please.

Have hard-coded correspondingly
